front/app/containers/Admin/pagesAndMenu/containers/GenericHeroBannerForm/BannerImageFields/index.tsx

```diff
@@ -197,77 +179,29 @@ const BannerImageField = ({
             />
           </Box>
         )}
-        {displayImageCropper ? (
-          <Box display="flex" flexDirection="column" gap="8px">
-            <ImageCropper
-              image={headerLocalDisplayImage}
-              onComplete={onAddImage}
-            />
-            <Warning>
-              <Text>
-                <FormattedMessage
-                  {...messages.fixedRatioImageCropperInfo}
-                  values={{
-                    link: (
-                      <a
-                        href={formatMessage(messages.imageSupportPageURL)}
-                        target="_blank"
-                        rel="noreferrer"
-                      >
-                        <FormattedMessage
-                          {...messages.fixedRatioImageCropperInfoLink}
-                        />
-                      </a>
-                    ),
-                  }}
-                />
-              </Text>
-            </Warning>
-          </Box>
-        ) : (
-          <HeaderImageDropzone
-            onAdd={bannerImageAddHandler}
-            onRemove={bannerImageRemoveHandler}
-            overlayColor={bannerOverlayColor}
-            overlayOpacity={bannerOverlayOpacity}
-            headerError={bannerError}
-            header_bg={headerLocalDisplayImage}
-            previewDevice={previewDevice}
-            layout={bannerLayout || 'full_width_banner_layout'}
+        <ImageUploader
+          bannerLayout={bannerLayout}
+          bannerOverlayColor={bannerOverlayColor}
+          bannerOverlayOpacity={bannerOverlayOpacity}
+          displayImageCropper={displayImageCropper}
+          displayOverlayControls={displayOverlayControls}
+          onAddImage={onAddImage}
+          onAddImageToUploader={handleOnAddImageToUploader}
+          onRemoveImageFromUploader={handleOnRemoveImageFromUploader}
+          previewDevice={previewDevice}
+          bannerError={bannerError}
+          headerLocalDisplayImage={headerLocalDisplayImage}
+        />
+        {/* We only allow the overlay for the full-width and fixed-ratio banner layout for the moment. */}
+        {displayOverlayControls && (
+          <OverlayControls
+            bannerOverlayColor={bannerOverlayColor}
+            onOverlayColorChange={onOverlayColorChange}
+            bannerOverlayOpacity={bannerOverlayOpacity}
+            onOverlayOpacityChange={onOverlayOpacityChange}
```

Comment:
Way too many props. In the future we'll have 1 handler and pass all there values via 1 object, which should mean 5-6 props fewer already.